agent/consul/leader.go

```diff
@@ -635,6 +635,38 @@ func (s *Server) upsertManagementToken(name, secretID string) error {
 	return nil
 }
 
+func (s *Server) deleteManagementToken(secretId string) error {
+	state := s.fsm.State()
+
+	// Fetch the token to get its accessor ID and to verify that it's a management token
+	_, token, err := state.ACLTokenGetBySecret(nil, secretId, nil)
+	if err != nil {
+		return fmt.Errorf("failed to get management token: %v", err)
+	}
+
+	if token == nil {
+		// token is already deleted
+		return nil
+	}
+
+	accessorID := token.AccessorID
+	if len(token.Policies) != 1 && token.Policies[0].ID != structs.ACLPolicyGlobalManagementID {
+		return fmt.Errorf("failed to delete %s: not a management token", token.Description)
```

Comment:
Wondering if we could be leaking a token description here in the case this error is returned? Not sure how much a problem that is, but wanted to call it out.

Reply:
I'm also unsure if it would be an issue to return that description, so I'm good to remove it from the error!